lib/Migration/Version20002Date20220201085856.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Gestion\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\IDBConnection;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version20002Date20220201085856 extends SimpleMigrationStep {
+	/**
+	 * IDB
+	 */
+	private $rows = [];
+    private IDbConnection $db;
+
+    public function __construct(IDbConnection $db){
+        $this->db = $db;
+    }
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		$schema = $schemaClosure();
+		$tableprefix = "gestion_";
+		if($schema->getTable($tableprefix.'client')->hasColumn('siret')) {
+			$query = $this->db->getQueryBuilder();
+			$query	->select('id', 'siret')
+					->from($tableprefix.'client');
+			$this->rows = $query->execute()->fetchAll();
+		}
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$schema = $schemaClosure();
+		$tableprefix = "gestion_";
+
+		$table = $schema->getTable($tableprefix.'client');
+
+		if ($table->hasColumn('siret')) {
+			$table->dropColumn('siret');
+		}
+
+		if (!$table->hasColumn('legal_one')) {
+            $table->addColumn('legal_one', 'text', []);
+        }
+
+		$table = $schema->getTable($tableprefix.'configuration');
+		if (!$table->hasColumn('legal_one')) {
+            $table->addColumn('legal_one', 'text', []);
+        }
+        if (!$table->hasColumn('legal_two')) {
+            $table->addColumn('legal_two', 'text', []);
+        }
+
+		return $schema;
+
+
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		foreach ($this->rows as $row) {
+     		$stmt = $this->db->prepare("UPDATE `".'*PREFIX*'."gestion_client` SET `legal_one` = ? WHERE id = ?;");
+        	$stmt->execute(array($row['siret'], $row['id']));
```

Comment:
You can use queryBuilder to don't get problem with different databases:

```suggestion
			$qb = $this->db->getQueryBuilder();
			$qb
				->update('gestion_client')
				->set('legal_one', $qb->createNamedParameter($row['siret']))
				->where($qb->expr()->eq('id', $qb->createNamedParameter($row['id'])))
				->execute();
```

Reply:
I don't know what that do : 
(it's why I don't use) Could you explain ?

```
$qb->createNamedParameter($row['siret'])

Reply:
Is the QueryBuilder:
https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/query-builder.html

You will generate the query using PHP methods and when the `execute` method was executed, inside of `execute` all parts of query will be converted to SQL syntax compatible with current database.

Example.... If you need use `limit` on a query, Oracle have an specific syntax for limit. You don't need know this, you only will call the method `limit` and the QueryBuilder will do all rest of magic.

The `createNamedParameter` is to pass a parameter to the query. Is good run queries using parameters to optimize the execution plan of current database. Like that you already did on first code passing the parameters to method execute and using `?` to define the place of parameter. But, named parameter will put the parameter with name on query.

Reply:
After merge this PR, test the code. I only copied from other place and adapted to this necessity.

Reply:
Ok